src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SyntaxVisualizerControl.xaml.cs

```diff
@@ -316,6 +316,21 @@ private void NavigateToCaret()
 
         var caret = _activeWpfTextView.Selection.StreamSelectionSpan.SnapshotSpan.Span.Start;
 
+        var codeDocument = GetCodeDocument();
+        if (codeDocument is null)
+        {
+            return;
+        }
+
+        var tree = codeDocument.GetSyntaxTree();
+
+        var change = new SourceChange(caret, 0, string.Empty);
+        // Sadly introducing the namespace to allow us to call the extension method causes heaps of type name conflicts
+        var owner = AspNetCore.Razor.Language.Legacy.LegacySyntaxNodeExtensions.LocateOwner(tree.Root, change);
+
+        var ownerString = owner is null ? "(null)" : $"{owner.Kind} [{owner.SpanStart}-{owner.Span.End}]";
```

Comment:
🤷‍♂️

According to [the docs](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Syntax/SyntaxNode.cs,115) there is a slight perf benefit of `SpanStart` over `Span.Start`, and I guess `SpanEnd` wasn't needed by the compiler, because it doesn't exist. This part of the Roslyn API has always seemed a bit odd to me though, too :)

Reply:
The short answer is that `SpanEnd` would perform exactly the same computation that is performed to produce `Span`. So, it wasn't added. However, `SpanStart` was deemed an improvement enough to warrant the extra API to avoid unnecessary computation when just the start position is needed. Honestly, in this case, it's really not worth it. Since you're accessing both the start and end of the span, it's really more efficient to pull `owner.Span` into a local and just call `Start` and `End`. Of course, the efficiency is pretty negligible.